Data/Tree.hs

```diff
@@ -21,20 +21,38 @@
 -- Maintainer  :  libraries@haskell.org
 -- Portability :  portable
 --
--- Multi-way trees (/aka/ rose trees) and forests.
+-- = Multi-way (aka. Rose Trees) and Forests
+--
+-- The @'Tree' a@ type represents a lazy, possibly infinite, multi-way tree
+-- (/aka./ rose tree).
+--
+-- The @'Forest' a@ type represents a forest of @'Tree' a@s.
 --
```

Comment:
I'm somewhat more inclined to stop using the `Forest` type synonym, and explain somewhere that a "forest" is a list of trees. What do you think?

Reply:
I tend to agree, this definition doesn't even match what anyone who's familiar with the term forest would expect either since a forest from graph theory is an undirected graph where each connected component is a tree. That being said, I don't think this PR is the place to remove it :)